plugins/janus_videoroom.c

```diff
@@ -2208,6 +2189,68 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
 	}
 }
 
+static void janus_videoroom_recorder_create(janus_videoroom_participant *participant, gboolean audio, gboolean video) {
```

Comment:
So called "recursive" or "re-entrant" mutexes can deal with that, but the default kind of mutexes cannot.
(You can initialize a pthread mutex with an attr `PTHREAD_MUTEX_RECURSIVE`, but I'd recommend not starting to use recursive mutexes in this project).